pkg/pillar/pubsub/pubsub.go

```diff
@@ -62,6 +64,30 @@ type PubSub struct {
 	updaterList *Updaters
 }
 
+// PublicationOptions defines all the possible options a new publication may have
+type PublicationOptions struct {
+	AgentName  string
+	AgentScope string
+	TopicType  interface{}
+	Persistent bool
+}
+
+// NewPublication creates a new publication with given options
+func NewPublication(options PublicationOptions) (Publication, error) {
+	defaultPubSub := New(&socketdriver.SocketDriver{})
+	if options.Persistent && options.AgentName != "" && options.TopicType != nil {
+		return defaultPubSub.PublishPersistent(options.AgentName, options.TopicType)
+	}
+	if options.AgentScope != "" && !options.Persistent && options.AgentName != "" && options.TopicType != nil {
+		return defaultPubSub.PublishScope(options.AgentName, options.AgentScope, options.TopicType)
+	}
+	if !options.Persistent && options.AgentName != "" && options.TopicType != nil {
+		return defaultPubSub.Publish(options.AgentName, options.TopicType)
+	}
```

Comment:
Hmm.. Each of these If statements must be simplified..

Basically, you have the following - right?
     if persistent {
         // Check conditions .. If all satisfy, return Ppropriate type
         // Else - Return Error
     }
      // Else - Non Persistent
       if options.AgentScope != "" {
       } else {
       }
    
etc.. Something like this.. It is not very clear to read code with these complex if-statements